californium-core/src/main/java/org/eclipse/californium/core/observe/InMemoryObservationStore.java

```diff
@@ -20,27 +20,28 @@
 import org.eclipse.californium.core.network.Exchange.KeyToken;
 import org.eclipse.californium.core.network.serialization.DataParser;
 import org.eclipse.californium.core.network.serialization.Serializer;
+import org.eclipse.californium.elements.CorrelationContext;
 import org.eclipse.californium.elements.RawData;
 
-public class InMemoryObserveRequestStore implements ObserveRequestStore {
+public class InMemoryObservationStore implements ObservationStore {
 
-	private Map<KeyToken, Request> map = new ConcurrentHashMap<>();
+	private Map<KeyToken, Observation> map = new ConcurrentHashMap<>();
 
 	@Override
-	public void add(Request request) {
-		if (request != null) {
-			map.put(new KeyToken(request.getToken()), request);
+	public void add(Observation obs) {
+		if (obs != null) {
+			map.put(new KeyToken(obs.getRequest().getToken()), obs);
 		}
 	}
 
 	@Override
-	public Request get(byte[] token) {
-		Request request = map.get(new KeyToken(token));
-		if (request != null) {
-			RawData serialize = Serializer.serialize(request, null);
+	public Observation get(byte[] token) {
```

Comment:
`KeyToken` is an internal object used to allow a byte array as key in a map. I'm not sure this is a good idea to expose it the `observationStore` API just to avoid to create a keyToken object in `InMemoryObservationStore`.